components/eamxx/src/share/grid/remap/coarsening_remapper.cpp

```diff
@@ -347,11 +414,119 @@ void CoarseningRemapper::do_remap_fwd ()
         "  - send rank: " + std::to_string(m_comm.rank()) + "\n");
   }
 
+  // Rescale any fields that had the mask applied.
+  for (int i=0; i<m_num_fields; ++i) {
+    const auto& f_ov_tgt = m_ov_tgt_fields[i];
+  }
+  if (m_track_mask) {
+    for (int i=0; i<m_num_fields; ++i) {
+      const auto& f_tgt = m_tgt_fields[i];
+      const int   mask_idx = m_mask_map_src.at(f_tgt.name());
+      if (mask_idx != -1) {
+        // Then this field did use a mask
+        const int mask_f_idx = m_mask_map_tgt.at(mask_idx);
+        const auto& mask = m_tgt_fields[mask_f_idx];
+        const auto  rank = f_tgt.get_header().get_identifier().get_layout().rank();
+        const auto& tgt_ap = f_tgt.get_header().get_alloc_properties();
+        if (can_pack && tgt_ap.is_compatible<RPack<16>>()) {
+          rescale_masked_fields<16>(f_tgt,mask);
+        } else if (can_pack && tgt_ap.is_compatible<RPack<8>>()) {
+          rescale_masked_fields<8>(f_tgt,mask);
+        } else if (can_pack && tgt_ap.is_compatible<RPack<4>>()) {
+          rescale_masked_fields<4>(f_tgt,mask);
+        } else {
+          rescale_masked_fields<1>(f_tgt,mask);
+        }
+      }
+    }
+  }
+}
+
+template<int PackSize>
+void CoarseningRemapper::
+rescale_masked_fields (const Field& x, const Field& mask) const
+{
+  using RangePolicy = typename KT::RangePolicy;
+  using MemberType  = typename KT::MemberType;
+  using ESU         = ekat::ExeSpaceUtils<typename KT::ExeSpace>;
+  using Pack        = ekat::Pack<Real,PackSize>;
+  using PackInfo    = ekat::PackInfo<PackSize>;
+
+  const auto& layout = x.get_header().get_identifier().get_layout();
+  const int rank = layout.rank();
+  const int ncols = m_ov_tgt_grid->get_num_local_dofs();
+  const Real mask_val = -999999.0;   // TODO: Maybe not hard code what a fully masked value will be set too.
+  const Real mask_threshold = 1e-8;  // TODO: Should we not hardcode the threshold for simply masking out the column.
```

Comment:
We certainly can set it to 0.  I don't have a reason to choose 1e-8, that's for sure.  The argument for caring about a threshold at all is to avoid say a single column in a map of 60 to dominate the whole column as though it shouldn't be masked.  For example, say column 1-60 map to column A in the coarse mesh.  If 2-60 are masked, and column 1 has a weight of 1e-12, we may want to see this and say that this is actually a masked column fully, throwing away the incredibly small 1e-12 value rather than scale it up to 1.0.

In a real life example, that would be like a columns 2-60 are inside a mountain, but column 1 is just on the edge, so not masked but barely adding anything.  Do we want the coarse mesh to take that value anyway?

The problem is setting the threshold though.  So hard-coding to 0.0 for now is probably the best choice and letting it be changed later  might be okay.

Reply:
I see your point. Though 1e-7 is prob also a similar scenario. Perhaps a reasonable rule of thumb could be 1/N, with being the number of nonzeros for that row (which is fixed, and same for all fields, so can be precomputed). It would mean that the non-masked cols add up to more than a single contribution of a uniform average of the patch.

But yeah, we can leave 0, and decide this factor later.